src/tests/src/solver/modeler/api/testModelerLPbuilder.cpp

```diff
@@ -106,10 +108,32 @@ BOOST_FIXTURE_TEST_CASE(three_fillers_given_to_builder___3_vars_3_constr_are_bui
     fillers = {oneVarFiller.get(), oneConstrFiller.get(), twoVarsTwoConstrFiller.get()};
 
     LinearProblemBuilder lpBuilder(fillers);
-    lpBuilder.build(*pb, LP_Data);
+    lpBuilder.build(*pb, LP_Data, ctx);
 
     BOOST_CHECK_EQUAL(pb->variableCount(), 3);
     BOOST_CHECK_EQUAL(pb->constraintCount(), 3);
 }
 
+BOOST_FIXTURE_TEST_CASE(FillerWithContext, Fixture)
+{
+    auto varFiller = std::make_unique<VarFillerContext>();
+    fillers = {varFiller.get()};
+
+    ctx = FillContext(0, 5);
+
+    ctx.scenariosSelected.push_back(0);
+    ctx.scenariosSelected.push_back(2);
+
+    LinearProblemBuilder lpBuilder(fillers);
+    lpBuilder.build(*pb, LP_Data, ctx);
+
+    BOOST_CHECK_EQUAL(pb->variableCount(), 10); // 5 timestep * 2 scenario
+
+    auto var1 = pb->getVariable("0-0");
```

Comment:
Please use a clearer naming here
```suggestion
    auto var1 = pb->getVariable("variable-ts0-sc0");
```